v2/clients/http/event.go

```diff
@@ -27,11 +31,86 @@ func NewEventClient(baseUrl string) interfaces.EventClient {
 	}
 }
 
-func (ec *eventClient) Add(ctx context.Context, reqs []requests.AddEventRequest) ([]common.BaseWithIdResponse, errors.EdgeX) {
+func (ec *eventClient) Add(ctx context.Context, reqs []requests.AddEventRequest) (
+	[]common.BaseWithIdResponse, errors.EdgeX) {
 	var br []common.BaseWithIdResponse
 	err := utils.PostRequest(ctx, &br, ec.baseUrl+v2.ApiEventRoute, reqs)
 	if err != nil {
 		return br, errors.NewCommonEdgeXWrapper(err)
 	}
 	return br, nil
 }
+
+func (ec *eventClient) AllEvents(ctx context.Context, offset, limit int) (
+	res responses.MultiEventsResponse, err errors.EdgeX) {
+	requestParams := url.Values{}
+	requestParams.Set(v2.Offset, strconv.Itoa(offset))
+	requestParams.Set(v2.Limit, strconv.Itoa(limit))
+	err = utils.GetRequest(ctx, &res, ec.baseUrl, v2.ApiAllEventRoute, requestParams)
+	if err != nil {
+		return res, errors.NewCommonEdgeXWrapper(err)
+	}
+	return res, nil
+}
+
+func (ec *eventClient) EventCount(ctx context.Context) (res common.CountResponse, err errors.EdgeX) {
+	err = utils.GetRequest(ctx, &res, ec.baseUrl, v2.ApiEventCountRoute, nil)
```

Comment:
Currently, json.Unmarshal is used in the request helper methods to parse the JSON-encoded response body into a specified response struct.
Refer to [https://golang.org/pkg/encoding/json/#Unmarshal](https://golang.org/pkg/encoding/json/#Unmarshal): `Unmarshal parses the JSON-encoded data and stores the result in the value pointed to by v. If v is nil or not a pointer, Unmarshal returns an InvalidUnmarshalError.`
Therefore, I think passing pointer from here could make the code concise.

Or we can make the request helper methods just return the response body in bytes so that there is no need to pass pointer. However, we still need json.Unmarshal function for unmarshaling...

@lenny-intel @cloudxxx8 any suggestions?

Reply:
Ahh, OK, that makes sense. I think using the name return variable hides this. How about changing to:
```
func (ec *eventClient) EventCount(ctx context.Context) (common.CountResponse, errors.EdgeX) {
       response := common.CountResponse{}
	err := utils.GetRequest(ctx, &res, ec.baseUrl, v2.ApiEventCountRoute, nil)
```

Reply:
Modified.